compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -693,9 +693,13 @@ execMigrate projectOpts inFile1_ inFile2_ mbDir =
                   let dar = ZipArchive.toArchive $ BSL.fromStrict bytes
                   -- get the main pkg
                   dalfManifest <- either fail pure $ readDalfManifest dar
-                  let pkgName = takeBaseName $ mainDalfPath dalfManifest
                   mainDalfEntry <- getEntry (mainDalfPath dalfManifest) dar
-                  (mainPkgId, mainLfPkg) <- decode $ BSL.toStrict $ ZipArchive.fromEntry mainDalfEntry
+                  (mainPkgId, mainLfPkg) <-
+                      decode $ BSL.toStrict $ ZipArchive.fromEntry mainDalfEntry
+                  let baseName = takeBaseName $ mainDalfPath dalfManifest
+                  let pkgName =
```

Comment:
When do we expect this to be `Nothing`? If the answer is “never” we should probably just throw an exception instead of trying to default to something.

Reply:
That would be backwards incompatible with old dars, but considering that you can't import cross sdk right now anyways, I think to error out here.